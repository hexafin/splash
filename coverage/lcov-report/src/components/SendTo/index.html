<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SendTo</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/SendTo
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.76% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>44/153</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.92% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>48/166</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">26.09% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/46</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.05% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>43/148</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Hits.js"><a href="Hits.js.html">Hits.js</a></td>
	<td data-value="43.48" class="pic low"><div class="chart"><div class="cover-fill" style="width: 43%;"></div><div class="cover-empty" style="width:57%;"></div></div></td>
	<td data-value="43.48" class="pct low">43.48%</td>
	<td data-value="23" class="abs low">10/23</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="8" class="abs low">3/8</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	</tr>

<tr>
	<td class="file medium" data-value="SearchBox.js"><a href="SearchBox.js.html">SearchBox.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="SendButton.js"><a href="SendButton.js.html">SendButton.js</a></td>
	<td data-value="42.86" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="SendLineItem.js"><a href="SendLineItem.js.html">SendLineItem.js</a></td>
	<td data-value="63.64" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 63%;"></div><div class="cover-empty" style="width:37%;"></div></div></td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="11" class="abs medium">7/11</td>
	<td data-value="76.47" class="pct medium">76.47%</td>
	<td data-value="17" class="abs medium">13/17</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="11" class="abs medium">7/11</td>
	</tr>

<tr>
	<td class="file low" data-value="SendTo.js"><a href="SendTo.js.html">SendTo.js</a></td>
	<td data-value="19.63" class="pic low"><div class="chart"><div class="cover-fill" style="width: 19%;"></div><div class="cover-empty" style="width:81%;"></div></div></td>
	<td data-value="19.63" class="pct low">19.63%</td>
	<td data-value="107" class="abs low">21/107</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="132" class="abs low">33/132</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="28" class="abs low">4/28</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="105" class="abs low">21/105</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
