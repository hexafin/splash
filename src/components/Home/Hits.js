import React, {Component} from "react"
import { connectInfiniteHits } from 'react-instantsearch/connectors';
import { StyleSheet, FlatList } from 'react-native';
import SplashtagButton from "./SplashtagButton"

const Hits = connectInfiniteHits(({ hits, hasMore, refine, callback }) => {

  /* if there are still results, you can
  call the refine function to load more */
  const onEndReached = function() {
    if (hasMore) {
      refine();
    }
  };

  return (
    <FlatList
      data={hits}
      onEndReached={onEndReached}
      keyExtractor={(item, index) => item.objectID}
      renderItem={({ item }) => {
        return (
          <SplashtagButton
            user={item}
            onPress={() => callback(item)}/>
        )}}
       />       
  );
});

export default Hits