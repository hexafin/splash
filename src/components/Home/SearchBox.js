import React from "react"
import {
    StyleSheet,
    View,
    TextInput,
    Text
} from "react-native"
import {colors} from "../../lib/colors"
import {
  connectInfiniteHits,
  connectSearchBox,
} from 'react-instantsearch/connectors';


const SearchBox = connectSearchBox(({ refine, currentRefinement }) => {

    return (
		<View style={styles.splashtagInputWrapper}>
			<Text style={styles.inputPrefix}>@</Text>
			<TextInput
			    onChangeText={text => refine(text)}
			    value={currentRefinement}
   				autoCapitalize={"none"}
				placeholder={"yourfriend"}
				style={styles.splashtagInput}
		        autoCorrect={false}
	        />
        </View>
    )
})

const styles = StyleSheet.create({
	inputPrefix: {
		fontSize: 22,
		fontWeight: "600",
		color: colors.gray,
		position: "absolute",
		left: 20,
		top: 20
	},
	splashtagInput: {
		padding: 20,
		fontSize: 22,
		fontWeight: "600"
	},
	splashtagInputWrapper: {
		shadowOffset: {
            width: 0,
            height: 5,
        },
        shadowOpacity: 0.05,
        shadowRadius: 12,
        marginTop: 10,
        marginBottom: 15,
        borderRadius: 5,
        paddingLeft: 30,
        paddingRight: 30,
        backgroundColor: colors.white
	}
})

export default SearchBox