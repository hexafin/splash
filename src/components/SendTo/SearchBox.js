import React from "react";
import { StyleSheet, View, TextInput, Text } from "react-native";
import { colors } from "../../lib/colors";
import {
	connectInfiniteHits,
	connectSearchBox
} from "react-instantsearch/connectors";

export const SearchBox = ({ refine, currentRefinement, onChange }) => {
	return (
		<View style={styles.inputWrapper}>
			<TextInput
				onChangeText={text => {
					refine(text);
					onChange(text);
				}}
				value={currentRefinement}
				autoCapitalize={"none"}
				placeholder={"Splashtag, Phone or Address"}
				style={styles.input}
				autoCorrect={false}
			/>
		</View>
	);
};

const styles = StyleSheet.create({
	input: {
		padding: 20,
		fontSize: 20,
		fontWeight: "500",
		color: colors.darkGray
	},
	inputWrapper: {
		shadowOffset: {
			width: 0,
			height: 5
		},
		shadowOpacity: 0.05,
		shadowRadius: 12,
		marginTop: 10,
		marginBottom: 15,
		borderRadius: 5,
		// paddingLeft: 50,
		backgroundColor: colors.white
	}
});

export default connectSearchBox(SearchBox);
