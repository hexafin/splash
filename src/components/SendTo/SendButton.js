import React, { Component } from "react";
import { Animated, View, Text, TouchableWithoutFeedback, Image, StyleSheet } from "react-native";
import { colors } from "../../lib/colors";
import ReactNativeHapticFeedback from "react-native-haptic-feedback";

/*
Button in SendTo page with option for finding recipient
<SendButton
	image={image}
	title={text}
	onPress={function}
/>
*/

class SendButton extends Component {
	componentWillMount() {
		this.scale = new Animated.Value(1);
	}
	render() {
		return (
			<TouchableWithoutFeedback
				onPressIn={() => {
					ReactNativeHapticFeedback.trigger("impactLight", true);
					Animated.spring(this.scale, {
						toValue: 0.8
					}).start();
				}}
				onPressOut={() => {
					ReactNativeHapticFeedback.trigger("impactLight", true);
					Animated.spring(this.scale, {
						toValue: 1.0
					}).start();
				}}
				onPress={this.props.onPress}
			>
				<Animated.View
					style={[
						styles.wrapper,
						{
							transform: [{ scale: this.scale }]
						}
					]}
				>
					<Image source={this.props.image} style={styles.image} resizeMode="contain" />
					<Text style={styles.title}>{this.props.title}</Text>
				</Animated.View>
			</TouchableWithoutFeedback>
		);
	}
}

export default SendButton;

const styles = StyleSheet.create({
	wrapper: {
		flexDirection: "row",
		justifyContent: "center",
		alignItems: "center",
		width: 150,
		height: 65,
		borderRadius: 3,
		backgroundColor: colors.primaryLight2,
		shadowOffset: {
			width: 0,
			height: 10
		},
		shadowOpacity: 0.1,
		shadowColor: colors.shadowPrimary,
		shadowRadius: 12
	},
	image: {
		width: 25,
		height: 25,
		marginRight: 15
	},
	title: {
		color: colors.white,
		// opacity: 0.9,
		fontSize: 16,
		textAlign: "left",
		fontWeight: "600"
	}
});
