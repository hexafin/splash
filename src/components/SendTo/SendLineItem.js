import React, {Component} from "react"
import {
	Animated,
	View,
	Text,
	TouchableWithoutFeedback,
	Image,
	StyleSheet,
} from "react-native"
import {colors} from "../../lib/colors"
import ReactNativeHapticFeedback from "react-native-haptic-feedback"

class SendLineItem extends Component {
	componentWillMount() {
		this.scale = new Animated.Value(1)
	}
	render() {

		const {
			selected=false,
			title,
			subtitle,
			circleText=null,
			address=null,
			extraContent=null,
			verified=true,
		} = this.props

		const animatedScale = {
			transform: [{scale: this.scale}]
		}

		let wrapperStyles = [styles.wrapperBase, animatedScale]
		wrapperStyles.push(selected ? styles.wrapperSelected : styles.wrapperUnselected)

		return (
			<TouchableWithoutFeedback onPressIn={() => {
				ReactNativeHapticFeedback.trigger("impactLight", true)
		        Animated.spring(this.scale, {
		            toValue: .8
		        }).start()
			}} onPressOut={() => {
				ReactNativeHapticFeedback.trigger("impactLight", true)
		        Animated.spring(this.scale, {
		            toValue: 1.0
		        }).start()
			}} onPress={this.props.onPress}>
				<Animated.View style={wrapperStyles}>

					<View style={styles.row}>
						<View style={styles.circle}>
							{circleText && <Text style={styles.circleText}>{circleText}</Text>}
							{!circleText &&
								<Image
									style={styles.circleSplash} 
									resizeMode="contain" 
									source={require("../../assets/icons/primarySplash.png")}/>}
						</View>
						<View style={styles.column}>
							<View style={styles.row}>
								<Text style={styles.title}>{title}</Text>
								{verified && <Image
									style={styles.verified}
									resizeMode="contain"
									source={require("../../assets/icons/checkmarkGreen.png")}/>}
							</View>
							<Text style={styles.subtitle}>{subtitle}</Text>
						</View>
					</View>
					{address && <Text style={styles.address}>{address}</Text>}
					{extraContent && <Text style={styles.extraContent}>{extraContent}</Text>}
				</Animated.View>
			</TouchableWithoutFeedback>
		)
	}
}

export default SendLineItem

const styles = StyleSheet.create({
	wrapperBase: {
		flexDirection: "column",
		justifyContent: "center",
		alignItems: "flex-start",
		padding: 15,
		marginTop: 10,
		borderRadius: 5,
		backgroundColor: colors.white,
		borderWidth: 1,
		borderColor: colors.white,
	},
	wrapperSelected: {
		borderWidth: 1,
		borderColor: colors.primary,
	},
	wrapperUnselected: {
		shadowOffset: {
			width: 0,
			height: 5,
		},
		shadowOpacity: 0.08,
		shadowColor: colors.shadowPrimary,
		shadowRadius: 10,
	},
	circle: {
		width: 40,
		height: 40,
		borderRadius: 20,
		marginRight: 10,
		backgroundColor: colors.primaryLight,
		justifyContent: "center",
		alignItems: "center",
	},
	circleText: {
		color: colors.primary,
		fontSize: 18,
		fontWeight: "600",
	},
	circleSplash: {
		width: 25,
		height: 25,
	},
	verified: {
		width: 18,
		height: 18,
		marginLeft: 5,
	},
	row: {
		flexDirection: "row",
		justifyContent: "flex-start",
		alignItems: "center",
	},
	column: {
		flexDirection: "column",
		justifyContent: "center",
		alignItems: "flex-start",
	},
	extraContent: {
		color: colors.gray,
		fontSize: 16,
		textAlign: "left",
		marginTop: 5,
	},
	address: {
		color: colors.primary,
		fontSize: 16,
		textAlign: "left",
		marginTop: 5,
	},
	title: {
		color: colors.darkGray,
		// opacity: 0.9,
		fontSize: 18,
		textAlign: "left",
		fontWeight: "700",
	},
	subtitle: {
		color: colors.gray,
		fontSize: 16,
		textAlign: "left",
		marginTop: 5,
	},
})