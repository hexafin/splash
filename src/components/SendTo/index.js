import SendTo from "./SendTo"
import {connect} from "react-redux"
import {bindActionCreators} from "redux"
import {sendTo} from "../../redux/payFlow"

const mapStateToProps = (state) => {
    return {
    	sendCurrency: state.payFlow.currency,
    	sendAmount: state.payFlow.amount,
    	bitcoinNetwork: state.crypto.wallets.BTC.network,
    	bitcoinAddress: state.crypto.wallets.BTC.address,
    	address: state.payFlow.address,
    	capturedQr: state.payFlow.capturedQr,
        userId: state.user.id,
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
    	sendTo
    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(SendTo)
