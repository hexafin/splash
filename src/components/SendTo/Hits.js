import React, {Component} from "react"
import { connectInfiniteHits } from 'react-instantsearch/connectors';
import {
	StyleSheet,
	FlatList,
	View,
} from 'react-native';
import SendLineItem from "./SendLineItem"

const Hits = connectInfiniteHits(({ hits, hasMore, refine, callback, userSplashtag }) => {

  /* if there are still results, you can
  call the refine function to load more */
  const onEndReached = function() {
    if (hasMore) {
      refine();
    }
  };

  return (
	    <FlatList
	      data={hits}
	      contentContainerStyle={styles.wrapper}
	      onEndReached={onEndReached}
	      keyExtractor={(item, index) => item.objectID}
	      renderItem={({ item }) => {
	      	console.log(item)
	      	if (item.splashtag != userSplashtag) {
	      		return (
		          <SendLineItem
		            title={`@${item.splashtag}`}
		            subtitle={item.phoneNumber}
		            circleText={item.splashtag.slice(0,2).toUpperCase()}
		            onPress={() => callback(item)}/>
		        )
	      	}
	      }}
	    />   
  );
});

const styles = StyleSheet.create({
	wrapper: {
		overflow: "visible",
		paddingHorizontal: 20,
	}
})

export default Hits