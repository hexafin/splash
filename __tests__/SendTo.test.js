import "react-native";
import React from "react";
import SendTo from "../src/components/SendTo/SendTo";
import { Hits } from "../src/components/SendTo/Hits";
import { SearchBox } from "../src/components/SendTo/SearchBox";
import SendButton from "../src/components/SendTo/SendButton";
import SendLineItem from "../src/components/SendTo/SendLineItem";

global.window = {};

it("Send To renders correctly", () => {
	const snap = shallow(
		<SendTo
			activeCryptoCurrency={"BTC"}
			sendCurrency={"USD"}
			sendAmount={5.0}
			network={"mainnet"}
			address={"testAddress"}
			capturedQr={false}
			userId={"testUserId"}
			contacts={[
				{
					splashtag: "userman",
					phoneNumber: "+19785012350",
					objectID: "uid",
					wallets: {
						BTC: "testAddress"
					}
				}
			]}
			showApproveModal={false}
			LoadTransactions={() => {
				return Promise.resolve();
			}}
			LoadContacts={() => {
				return Promise.resolve();
			}}
			DismissTransaction={() => {}}
			addContactsInfo={() => {}}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Hits renders correctly", () => {
	const snap = shallow(
		<Hits
			contacts={[
				{
					splashtag: "userman",
					phoneNumber: "+19785012350",
					objectID: "uid",
					wallets: {
						BTC: "testAddress"
					}
				}
			]}
			hits={[
				{
					splashtag: "userman"
				}
			]}
			hasMore={true}
			refine={false}
			callback={() => {}}
			userId={"testUserId"}
			selectedId={"selectedUserId"}
			currency={"USD"}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("Search Box renders correctly", () => {
	const snap = shallow(<SearchBox />);
	expect(snap).toMatchSnapshot();
});

it("SendButton renders correctly", () => {
	const snap = shallow(<SendButton />);
	expect(snap).toMatchSnapshot();
});

it("SendLineItem renders correctly", () => {
	const snap = shallow(<SendLineItem />);
	expect(snap).toMatchSnapshot();
});
