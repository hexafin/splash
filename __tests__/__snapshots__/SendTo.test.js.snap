// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits renders correctly 1`] = `
<FlatList
  contentContainerStyle={
    Object {
      "overflow": "visible",
      "paddingBottom": 130,
      "paddingHorizontal": 20,
    }
  }
  data={
    Array [
      Object {
        "splashtag": "userman",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`Search Box renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "marginBottom": 15,
      "marginTop": 10,
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 12,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Splashtag, Phone or Address"
    style={
      Object {
        "color": undefined,
        "fontSize": 20,
        "fontWeight": "500",
        "padding": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
</Component>
`;

exports[`Send To renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
        "paddingBottom": 40,
        "paddingTop": 60,
      }
    }
  >
    <CloseButton
      color="dark"
      onPress={[Function]}
    />
    <FlatBackButton
      color="dark"
      onPress={[Function]}
    />
    <Component
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Component
        style={
          Object {
            "color": undefined,
            "fontSize": 24,
            "fontWeight": "600",
            "marginBottom": 4,
          }
        }
      >
        Sending 
        Bitcoin
      </Component>
    </Component>
    <Component
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <Component
        style={
          Object {
            "color": "#919191",
            "fontSize": 18,
            "fontWeight": "500",
            "letterSpacing": 1,
            "marginBottom": 2,
          }
        }
      >
        AMOUNT
      </Component>
      <Component
        style={
          Object {
            "color": undefined,
            "fontSize": 20,
            "fontWeight": "700",
          }
        }
      >
        USD
         
        $
        0.05
      </Component>
    </Component>
    <CreateInstantSearch
      apiKey="ba602b59c39f9584a485db5c32a55048"
      appId="HL2AA4354B"
      indexName="Users"
      refresh={false}
      root={
        Object {
          "Root": [Function],
        }
      }
    >
      <Component
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#919191",
              "fontSize": 18,
              "fontWeight": "500",
              "letterSpacing": 1,
              "marginBottom": 2,
            }
          }
        >
          SEARCH
        </Component>
        <AlgoliaSearchBox(SearchBox)
          onChange={[Function]}
        />
      </Component>
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 20,
              "paddingHorizontal": 25,
              "width": 750,
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <HeaderButton
          image={
            Object {
              "testUri": "../../../src/assets/icons/copyPaste.png",
            }
          }
          onPress={[Function]}
          title="Paste
address"
        />
        <HeaderButton
          image={
            Object {
              "testUri": "../../../src/assets/icons/qrIcon.png",
            }
          }
          onPress={[Function]}
          title="Scan
QR-Code"
        />
      </AnimatedComponent>
      <Component
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#919191",
              "fontSize": 18,
              "fontWeight": "500",
              "letterSpacing": 1,
              "marginBottom": 2,
            }
          }
        >
          YOUR CONTACTS
        </Component>
      </Component>
      <Component
        style={
          Object {
            "flex": 1,
            "minHeight": 400,
          }
        }
      >
        <FlatList
          ListFooterComponent={
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "paddingVertical": 25,
                }
              }
            >
              <Component
                style={
                  Object {
                    "color": "#919191",
                    "fontSize": 16,
                    "marginBottom": 8,
                  }
                }
              >
                Added new numbers?
              </Component>
              <Component
                style={
                  Object {
                    "color": "#4547E0",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Refresh Contacts
              </Component>
            </TouchableOpacity>
          }
          contentContainerStyle={
            Object {
              "overflow": "hidden",
              "paddingBottom": 130,
              "paddingHorizontal": 20,
            }
          }
          data={
            Array [
              Object {
                "objectID": "uid",
                "phoneNumber": "+19785012350",
                "splashtag": "userman",
                "wallets": Object {
                  "BTC": "testAddress",
                },
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </Component>
    </CreateInstantSearch>
    <NextButton
      disabled={true}
      onPress={[Function]}
      title="Preview Send"
    />
  </Component>
</TouchableWithoutFeedback>
`;

exports[`SendButton renders correctly 1`] = `
<TouchableWithoutFeedback
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#6163F4",
          "borderRadius": 3,
          "flexDirection": "row",
          "height": 65,
          "justifyContent": "center",
          "shadowColor": undefined,
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 12,
          "width": 150,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <Component
      resizeMode="contain"
      style={
        Object {
          "height": 25,
          "marginRight": 15,
          "width": 25,
        }
      }
    />
    <Component
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "left",
        }
      }
    />
  </AnimatedComponent>
</TouchableWithoutFeedback>
`;

exports[`SendLineItem renders correctly 1`] = `
<TouchableWithoutFeedback
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FFFFFF",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 10,
          "padding": 15,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
        Object {
          "shadowColor": undefined,
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 10,
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EFEFFD",
            "borderRadius": 20,
            "height": 40,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 40,
          }
        }
      >
        <Component
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/icons/primarySplash.png",
            }
          }
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Component
            style={
              Object {
                "color": undefined,
                "fontSize": 18,
                "fontWeight": "700",
                "textAlign": "left",
              }
            }
          />
          <Component
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/icons/checkmarkGreen.png",
              }
            }
            style={
              Object {
                "height": 18,
                "marginLeft": 5,
                "width": 18,
              }
            }
          />
        </Component>
        <Component
          style={
            Object {
              "color": "#919191",
              "fontSize": 16,
              "marginTop": 5,
              "textAlign": "left",
            }
          }
        />
      </Component>
    </Component>
  </AnimatedComponent>
</TouchableWithoutFeedback>
`;
